tests/Feature/Api/V1/UserCompletionProgressTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Api\V1;
+
+use App\Platform\Enums\UnlockMode;
+use App\Platform\Models\Achievement;
+use App\Platform\Models\Game;
+use App\Platform\Models\System;
+use App\Site\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Carbon;
+use Tests\Feature\Platform\Concerns\TestsPlayerAchievements;
+use Tests\Feature\Platform\Concerns\TestsPlayerBadges;
+use Tests\TestCase;
+
+class UserCompletionProgressTest extends TestCase
+{
+    use RefreshDatabase;
+    use BootstrapsApiV1;
+    use TestsPlayerAchievements;
+    use TestsPlayerBadges;
+
+    public function testItValidates(): void
+    {
+        $this->get($this->apiUrl('GetUserCompletionProgress'))
+            ->assertJsonValidationErrors([
+                'u',
+            ]);
+    }
+
+    public function testGetUserCompletionProgressUnknownUser(): void
+    {
+        $this->get($this->apiUrl('GetUserCompletionProgress', ['u' => 'nonExistant']))
+            ->assertSuccessful()
+            ->assertJson([]);
+    }
+
+    public function testGetUserCompletionProgress(): void
+    {
+        Carbon::setTestNow(Carbon::now());
+
+        /**
+         * Set up a user with 10 games played:
+         * - 2 unfinished
+         * - 3 beaten
+         * - 1 beaten (softcore)
+         * - 3 completed (one of which is also beaten softcore)
+         * - 4 mastered (two of which are also beaten hardcore)
```

Comment:
if a game is completed (softcore) and beaten (hardcore), which is reported?

Reply:
Similar to the Completion Progress page, the Completion (softcore) will be reported as opposed to the Beaten (hardcore) award.